clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskDefinitionCachingAgent.java

```diff
@@ -75,18 +92,45 @@ public String getAgentType() {
       }
       ListTaskDefinitionsResult listTaskDefinitionsResult = ecs.listTaskDefinitions(listTasksRequest);
       List<String> taskDefinitionArns = listTaskDefinitionsResult.getTaskDefinitionArns();
+
       if (taskDefinitionArns.size() == 0) {
         continue;
       }
-      for (String taskDefintionArn : taskDefinitionArns) {
-        TaskDefinition taskDefinition = ecs.describeTaskDefinition(new DescribeTaskDefinitionRequest().withTaskDefinition(taskDefintionArn)).getTaskDefinition();
+
+      Set<String> newTaskDefArns = new HashSet<>(taskDefinitionArns);
```

Comment:
newTaskDefArns could be renamed to uncachedTaskDefArns, and modify the right-hand side to something like:

`new HashSet<>(new HashSet(taskDefinitionArns).removeAll(newTaskDefArns))`;

Reply:
Talked it over - no changes required.